kademlia/src/main/java/ingsw/group1/kademlia/NodesComparator.java

```diff
@@ -0,0 +1,28 @@
+package ingsw.group1.kademlia;
+
+import java.util.Comparator;
+
+/**
+ * Compares Nodes based on the distance to a target node
+ */
+public class NodesComparator implements Comparator<Node<BinarySet>> {
```

Comment:
The Comparator interface is meant to represent a class able to compare two objects between themselves, using the compare method for something else isn't a very good practice. I'd define some sort of other interface or make the comparison method a static method somewhere else.

Reply:
I see why you would declare this as extending Comparator though, to use it in Collections.sort(), I suppose you could leave it like this since I don't see a better solution right now

Reply:
Exactly, it's not quite the cleanest thing, but it's handy